rust/src/lib.rs

```diff
@@ -2679,6 +2784,107 @@ pub unsafe extern "C" fn zcashlc_shield_funds(
     unwrap_exc_or(res, false)
 }
 
+#[no_mangle]
+pub unsafe extern "C" fn zcashlc_propose_shielding(
+    db_data: *const u8,
+    db_data_len: usize,
+    usk_ptr: *const u8,
+    usk_len: usize,
+    memo: *const u8,
+    shielding_threshold: u64,
+    network_id: u32,
+    min_confirmations: u32,
+    use_zip317_fees: bool,
+    proposal_bytes_ret: *mut u8
+) -> bool {
+    let res = catch_panic(|| {
+        let network = parse_network(network_id)?;
+        let min_confirmations = NonZeroU32::new(min_confirmations)
```

Comment:
The upstream APIs have changed to allow zero-conf shielding transactions to be created, so we should just make this a `u32`.

Reply:
Updated.